examples/partitionContainer/partition.py

```diff
@@ -0,0 +1,124 @@
+import inputParser
+import re
+import sys
+
+
+# This function is from yomi, can be found at https://github.com/openSUSE/yomi/blob/f3f4ac60852fa79665e77ede90960860ddd684fb/salt/_utils/disk.py#L31
+def units(value, default="MB"):
+    """
+    Split a value expressed (optionally) with units.
+
+    Returns the tuple (value, unit)
+    """
+    valid_units = (
+        "B",
+        "KB",
+        "kB",
+        "MB",
+        "MiB",
+        "GB",
+        "GiB",
+        "TB",
+        "TiB",
+        "%",
+    )
+    match = re.search(r"^([\d.]+)(\D*)$", str(value))
+    if match:
+        value, unit = match.groups()
+        unit = unit if unit else default
+        if unit in valid_units:
+            return float(value), unit
+        else:
+            raise Exception("{} not recognized as a valid unit".format(unit))
+    raise Exception("{} cannot be parsed".format(value))
+
+def convertToBytes(value, unit):
+    multipliers = {
+        "B": 1,
+        "kB": 2**10,
+        "KB": 10**3,
+        "MiB": 2**20,
+        "MB": 10**6,
+        "GiB": 2**30,
+        "GB": 10**9,
+        "TiB": 2**40,
+        "TB": 10**12
+    }
+    return value * multipliers.get(unit)
+
+
+# Setup
+
+filename = sys.argv[1]
+
+environment = Environment(True)
+
+storage = Storage(environment)
+storage.probe()
+staging = storage.get_staging()
+
+
+# Processes input, outputs a normalized list of devices and the initial gap
+devList, initial_gap = inputParser.processInput(filename)
+# print(devList, initial_gap)
+
+
+# Begins partitioning the regions at the initial gap
+startingPoint = convertToBytes(units(initial_gap)[0], units(initial_gap)[1])
+
+for device_name, device_info in devList.items():
+
+    # Creates a temporary device with the its name
+    tempDevice = Partitionable.find_by_name(staging, device_name)
+    tempDevice.remove_descendants()
+    # For now it automatically creates a GPT partition table
+    # Hope to supporst MSDOS in the future
+    gpt = tempDevice.create_partition_table(PtType_GPT)
+
+    # Separates the block size and block size unit
+    block_size, block_unit = units(device_info.get("blockSize", "512B"))
+    
+    # Converts the block size into bytes if not already in bytes
+    # Since libstorage uses bytes for its Regions
+    blockSizeBytes = convertToBytes(block_size, block_unit)
+
+    for partition in device_info.get("partitions", []):
+        # Separates the partition size value and its unit
+        part_size, part_unit =  units(partition["size"])
+        if part_unit != "%":
+            partSizeBytes = convertToBytes(part_size, part_unit)
+        else:
+            print("Does not support percentage unit currently")
+
+        # Code breaks if probing occurs, but using the percentage unit requires probing
+        # else:
+        #     storage.probe()
+        #     probed = storage.get_probed()
+        #     disks = probed.get_all_disks()
+        #     for d in disks:
+        #         print(d.get_name())
```

Comment:
This is not needed here anymore as we are doing probing on L59. Can we make percentages work now?